torchsnapshot/snapshot.py

```diff
@@ -270,13 +277,18 @@ def async_take(
         storage = url_to_storage_plugin_in_event_loop(
             url_path=path, event_loop=event_loop
         )
+
+        if not custom_tensor_prepare_func:
+            custom_tensor_prepare_func = lambda path, tensor, tracing : tensor
```

Comment:
Taking `tracing` as an argument is pretty brilliant.

Though in the long run we would want to automatically infer the output dtype for the users because (1) it's technically possible (2) the requirement for providing a tracing mode puts too much cognitive overhead on non-power users.

For now this is good. In the future, we can detect the presence of the tracing argument in the udf signature and automatically perform tracing if the argument is absent.

Reply:
Awesome!  I figured there was some torch magic that could be used.  

Note that with the `make_custom_tensor_prepare_func` factory function I put in the comments, the user doesn't have to worry about the signature of `custom_tensor_prepare_func`.  They pass in a boolean function that returns `True` if a path should be quantized, and they get a `custom_tensor_prepare_func` back out.  No direct interaction with `tracing` is required.